src/main/authentication/ssoAuthentication.ts

```diff
@@ -0,0 +1,82 @@
+import process from 'process';
+import { getSsoUserGroups } from '../helpers/graphApiHelper';
+import { AccountManagementRequests } from '../resources/requests/AccountManagementRequests';
+import { FRONTEND_URL } from '../helpers/envUrls';
+import config = require('config');
+const authenticationConfig = require('./authentication-config.json');
+
+const ssoClientId = process.env.SSO_CLIENT_ID
+    ? process.env.SSO_CLIENT_ID
+    : config.get('secrets.pip-ss-kv.SSO_CLIENT_ID');
+const ssoClientSecret = process.env.SSO_CLIENT_SECRET
+    ? process.env.SSO_CLIENT_SECRET
+    : config.get('secrets.pip-ss-kv.SSO_CLIENT_SECRET');
+const ssoMetadata = process.env.SSO_CONFIG_ENDPOINT
+    ? process.env.SSO_CONFIG_ENDPOINT
+    : config.get('secrets.pip-ss-kv.SSO_CONFIG_ENDPOINT');
+
+export const ssoOidcConfig = {
+    identityMetadata: ssoMetadata,
+    clientID: ssoClientId,
+    responseType: authenticationConfig.RESPONSE_TYPE,
+    responseMode: authenticationConfig.RESPONSE_MODE,
+    redirectUrl: FRONTEND_URL + '/sso',
+    allowHttpForRedirectUrl: true,
+    clientSecret: ssoClientSecret,
+    scope: 'openid profile email',
+};
+
+const accountManagementRequests = new AccountManagementRequests();
+
+export class SsoAuthentication {
+    public async determineUserRole(oid: string, accessToken: string): Promise<string> {
+        const userGroupsObject = await getSsoUserGroups(oid, accessToken);
+        const securityGroupMap = new Map<string, string>([
+            [process.env.SSO_SG_SYSTEM_ADMIN, 'SYSTEM_ADMIN'],
+            [process.env.SSO_SG_SUPER_ADMIN_CTSC, 'INTERNAL_SUPER_ADMIN_CTSC'],
+            [process.env.SSO_SG_SUPER_ADMIN_LOCAL, 'INTERNAL_SUPER_ADMIN_LOCAL'],
+            [process.env.SSO_SG_ADMIN_CTSC, 'INTERNAL_ADMIN_CTSC'],
+            [process.env.SSO_SG_ADMIN_LOCAL, 'INTERNAL_ADMIN_LOCAL'],
+        ]);
+        let userRole;
+
+        if (userGroupsObject?.value.length > 0) {
+            const userGroups = userGroupsObject.value;
+            securityGroupMap.forEach((value: string, key: string) => {
```

Comment:
Is there a convenience function that we can use instead of this like the java matchAny function instead? of manually looping through

Reply:
I have updated the code slightly using 'find'. This is still not as simple as you would like. They are kind of 2 arrays here. One is all the groups the user belongs to, and the other is a list of CaTH security groups. After we find the security group we also need to get the equivalent role for that group.